test/qpy_compat/test_qpy.py

```diff
@@ -820,8 +820,36 @@ def generate_v12_expr():
     return [index, shift]
 
 
-def generate_circuits(version_parts):
-    """Generate reference circuits."""
+def generate_replay_with_expression_substitutions():
+    """Circuits with parameters that have substituted expressions in the replay"""
+    a = Parameter("a")
+    b = Parameter("b")
+    a1 = a * 2
+    a2 = a1.subs({a: 3 * b})
+    qc = QuantumCircuit(1)
+    qc.rz(a2, 0)
+
+    return [qc]
+
+
+def generate_v13_fix_expr():
+    """Circuits that contain expressions and types new in QPY v13 (after fix)."""
+    from qiskit.circuit.classical import expr, types
+
+    float_expr = QuantumCircuit(name="float_expr")
+    with float_expr.if_test(expr.less(1.0, 2.0)):
+        pass
+
+    return [float_expr]
+
+
+def generate_circuits(version_parts, current_version, load_context=False):
+    """Generate reference circuits.
+
+    If load_context is True, avoid generating Pulse-based reference
+    circuits. For those circuits, load_qpy only checks that the cached
+    circuits can be loaded without erroring."""
```

Comment:
This change probably shouldn't be backported because load_context was only on main/2.0 as part of the pulse removal.

```suggestion
def generate_circuits(version_parts):
    """Generate reference circuits."""
```